include/tscore/ink_memory.h

```diff
@@ -617,3 +618,45 @@ struct ats_unique_buf_deleter {
 };
 using ats_unique_buf = std::unique_ptr<uint8_t[], ats_unique_buf_deleter>;
 ats_unique_buf ats_unique_malloc(size_t size);
+
+class c_str_view
+{
+public:
+  using size_type = std::string_view::size_type;
+
+  c_str_view() : c_str_view(nullptr, 0) {}
+  explicit c_str_view(const char *data, size_type length)
+    : c_str_view{
+        std::string_view{data, length}
+  }
+  {
+  }
+  explicit c_str_view(std::string_view sv) : sv_{sv}
+  {
+    ink_assert(sv_.data() == nullptr ? sv_.length() == 0 : sv_.data()[sv_.length()] == '\0');
```

Comment:
`sv_.data()[sv_.length()] == '\0'` could be considered UB if the string_view is not null terminated as this might be reading past a buffer, but since this is an assert anyway its fine if it crashes when called with a non-null terminated string_view.